checkmates/data_checks/checks/distribution_data_check.py

```diff
@@ -0,0 +1,155 @@
+            ...         "data_check_name": "DistributionDataCheck",
+            ...         "level": "error",
+            ...         "details": {"columns": None, "rows": None, "unsupported_type": "datetime"},
+            ...         "code": "TARGET_UNSUPPORTED_TYPE",
+            ...         "action_options": []
+            ...     }
+            ... ]
+        """
+        messages = []
+
+        if y is None:
+            messages.append(
+                DataCheckError(
+                    message="Data is None",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_IS_NONE,
+                    details={},
+                ).to_dict(),
+            )
+            return messages
+
+        y = infer_feature_types(y)
+        allowed_types = [
+            ww.logical_types.Integer.type_string,
+            ww.logical_types.Double.type_string,
+            ww.logical_types.Age.type_string,
+            ww.logical_types.AgeFractional.type_string,
+        ]
+        is_supported_type = y.ww.logical_type.type_string in allowed_types
+
+        if not is_supported_type:
+            messages.append(
+                DataCheckError(
+                    message="Target is unsupported {} type. Valid Woodwork logical types include: {}".format(
+                        y.ww.logical_type.type_string,
+                        ", ".join([ltype for ltype in allowed_types]),
+                    ),
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_TYPE,
+                    details={"unsupported_type": X.ww.logical_type.type_string},
+                ).to_dict(),
+            )
+            return messages
+
+        (
+            is_skew,
+            distribution_type,
+            skew_value,
+            coef,
+        ) = _detect_skew_distribution_helper(X)
+
+        if is_skew:
+            details = {
+                "distribution type": distribution_type,
+                "Skew Value": skew_value,
+                "Bimodal Coefficient": coef,
+            }
+            messages.append(
+                DataCheckWarning(
+                    message="Data may have a skewed distribution.",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.SKEWED_DISTRIBUTION,
+                    details=details,
+                    action_options=[
+                        DataCheckActionOption(
+                            DataCheckActionCode.TRANSFORM_TARGET,
+                            data_check_name=self.name,
+                            metadata={
+                                "is_skew": True,
+                                "transformation_strategy": "yeojohnson",
+                            },
+                        ),
+                    ],
+                ).to_dict(),
+            )
+        return messages
+
+
+def _detect_skew_distribution_helper(X):
```

Comment:
We should be passing in only numeric values of X. Currently, we're not doing any filters for X in this MR, which means that if X has non-numerical values, we're passing it all to the numpy and diptest functions, which I assume will cause issues. 
Additionally, seems like this function only takes 1 column at a time as the value for X. Let's add typing info for X so we can easily determine what data to pass in, and then add logic above to call this function for each numeric column in our dataframe